### ISSUE TITLE
Allow passing a value to `then` of the main Listr instance

### ISSUE THREAD

**Issue by okonet** [SamVerschueren/listr#37]
In the following example:

```js
new Listr(tasks, { concurrent })
                    .run({
                        errors: []
                    })
                    .then((res) => {
                        console.log(res)
                    })
                    .catch((error) => {
                        console.error(error)
                        process.exit(1)
                    })
```

the `res` is always `undefined` even then I resolve internal tasks with some value.

Or alternatively add `context` to the `then` so it can be accessed.

**Comment by okonet** [comment]
The more I think about it, the less I see what should be passed to `then` as a value besides the `context`. In this case, I'd expect it also to be accessible in `catch`. But here is there the limitation of `Promise` will strike I think :(

**Comment by SamVerschueren** [comment]
You're right, the context should be returned as result. We could also easily attach it to the error object `err.context`.

**Comment by sndrs** [comment]
any chance you could publish a version with this @SamVerschueren? was following the readme and couldn't work out why i was getting `undefined` in `run().then(console.log)` 😄

**Comment by SamVerschueren** [comment]
Thanks for reminding me @sndrs ! I had hoped that I could resolve more issues in the meantime but I'm quite busy lately. Released in `0.8.0`.

**Comment by sndrs** [comment]
thanks @SamVerschueren!
